samples/apps/logging/logging.cpp

```diff
@@ -498,29 +503,43 @@ namespace loggingapp
         size_t id;
         if (!http::get_query_value(parsed_query, "id", id, error_reason))
         {
-          return ccf::make_error(
-            HTTP_STATUS_BAD_REQUEST,
-            ccf::errors::InvalidQueryParameterValue,
-            std::move(error_reason));
+          auto response = nlohmann::json{{
+            "error",
+            {
+              {"code", ccf::errors::InvalidQueryParameterValue},
+              {"message", std::move(error_reason)},
+            },
+          }};
+
+          ctx.rpc_ctx->set_response(response, HTTP_STATUS_BAD_REQUEST);
+          return;
         }
 
         auto record = committed_records->get(id);
 
         if (record.has_value())
         {
-          return ccf::make_success(LoggingGet::Out{record.value()});
+          nlohmann::json response = LoggingGet::Out{record.value()};
+          ctx.rpc_ctx->set_response(response, HTTP_STATUS_OK);
+          return;
         }
 
-        return ccf::make_error(
-          HTTP_STATUS_BAD_REQUEST,
-          ccf::errors::ResourceNotFound,
-          fmt::format("No such record: {}.", id));
+        auto response = nlohmann::json{{
+          "error",
+          {
+            {"code", ccf::errors::ResourceNotFound},
+            {"message", fmt::format("No such record: {}.", id)},
+            {"current_txid", committed_records->get_current_txid().to_str()},
```

Comment:
That doesn't seem thread safe, it could be called while the indexer is updating current_txid. The strategy should lock around access.

Reply:
See https://github.com/microsoft/CCF/pull/4075

Reply:
https://github.com/microsoft/CCF/pull/4803/commits/e86dcfd56e6f4820536d833545d745b22c159835